drivers/usb/udc/udc_dwc2_vendor_quirks.h

```diff
@@ -172,6 +181,11 @@ static inline int usbhs_enable_core(const struct device *dev)
 {
 	NRF_USBHS_Type *wrapper = USBHS_DT_WRAPPER_REG_ADDR(0);
 
+	if (!k_sem_count_get(&vbus_ready_sem)) {
+		LOG_WRN("VBUS is not ready, block udc_enable()");
+	}
+
+	k_sem_take(&vbus_ready_sem, K_FOREVER);
```

Comment:
This is good enough for first enable. However if application does `udc_enable()`, `udc_disable()`, `udc_enable()` while the USB cable is connected just once and held in place, then the code will block claiming VBUS is not ready.

I think `k_event` might be more appropriate here. `usbhs_vbus_handler()` would be posting and clearing the event bit while `usbhs_enable_core()` would simply call `k_event_wait()` without event reset.

Reply:
Yes, we need to track the VBUS state and hence the event API is best suited for this action. Please check the implementation at https://github.com/jfischer-no/zephyr/pull/4